### ISSUE TITLE
`cargo test` breaking with `forbid(missing_docs)`

### ISSUE THREAD

**Issue by y86-dev** [rust-lang/rust#130218]
I tried this code:

```rust
//! Provides a noop.

#![forbid(missing_docs)]

/// Does nothing
pub fn noop() {}
```
in `lib.rs`.

I expected to see this happen: successful `cargo test` invocation.

Instead, this happened: `cargo test` errors:
```text
$ cargo test
   Compiling rust v0.0.0 (/tmp/noop)
error[E0453]: allow(missing_docs) incompatible with previous forbid

For more information about this error, try `rustc --explain E0453`.
error: could not compile `rust` (lib test) due to 1 previous error
```

### Meta

This was triggered today in my daily testing pipeline using nightly. `cargo check` reports no issues.

`rustc --version --verbose`:
```
rustc 1.83.0-nightly (0ee7cb5e3 2024-09-10)
binary: rustc
commit-hash: 0ee7cb5e3633502d9a90a85c3c367eccd59a0aba
commit-date: 2024-09-10
host: x86_64-unknown-linux-gnu
release: 1.83.0-nightly
LLVM version: 19.1.0
```
